### PR TITLE
fix(refactor): async await instead of Promise chain

### PR DESCRIPTION
**Pull request by Uzlopak**
Use async await for better readability.

### CODE REVIEW

**Comment by Uzlopak** [comment]
Yes, it is kind of a heavy change, but it does not contain any breaking change :)

File: src/fetch-wrapper.ts [hunk 1]
```diff
@@ -32,109 +14,46 @@ export default function fetchWrapper(
...
+  );
+
+  let fetchResponse: Response;
+
+  try {
+    fetchResponse = await fetch(requestOptions.url, {
+      method: requestOptions.method,
+      body,
+      redirect: requestOptions.request?.redirect,
+      // Header values must be `string`
```
Comment (wolfy1339) [inline src/fetch-wrapper.ts:40]: Can you move this comment up to the definition of `requestHeaders`

File: src/fetch-wrapper.ts [hunk 2]
```diff
@@ -146,14 +65,87 @@ export default function fetchWrapper(
...
+
+  octokitResponse.data = parseSuccessResponseBody
+    ? await getResponseData(fetchResponse)
+    : fetchResponse.body;
+
+  return octokitResponse;
 }
 
-async function getResponseData(response: Response) {
+function getResponseData(response: Response) {
```
Comment (wolfy1339) [inline src/fetch-wrapper.ts:148]: Is the function `getResponseData()` async?

File: src/fetch-wrapper.ts (hunk 1 above)
Comment (Uzlopak) [reply src/fetch-wrapper.ts:40]: Sure.

File: src/fetch-wrapper.ts (hunk 2 above)
Comment (wolfy1339) [reply src/fetch-wrapper.ts:148]: That is fine just like that. Maybe a simple comment explaining the choice

**Comment by Uzlopak** [comment]
@wolfy1339 
I adapted your requested changes but added the async for getResponseData. I will touch that function in a separate PR.

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**